### ISSUE TITLE
Nightflyer Bugs

### ISSUE THREAD

**Issue by EvadLhorg** [throneteki/throneteki#3076]
Nightflyer seems to have some buggy interactions due to the Shadow (X) cost. The X is determined by a prompt after bringing it out of shadows. 
It seems you can trigger Bowels of Casterly Rock to gain 2 gold BEFORE choosing the X to pay for Nightflyer. This shouldn´t be allowed.
It seems that you can reduce the cost of Nightflyer with The Water Gardens, which shouldn´t be allowed.

**Comment by ystros** [comment]
Yup, Nightflyer is implemented to simulate the `Shadow (X)` using an X ability cost once it's already out of shadows. This is the first report of an interaction where it matters.
